feat(MaticJS): adds zkevm client (#1444)
maticnetwork/matic-docs: docs/develop/ethereum-polygon/matic-js/zkevm/erc20.md

---

:::tip

See the [<ins>MaticJS SDK examples</ins>](https://github.com/maticnetwork/matic.js/tree/master/examples/hermez/erc20) on our official Github repo to interact with ERC20 tokens on the zkEVM network..

:::

## Calling ERC20 Method

`HermezClient` provides `erc20` method which helps you to interact with an **ERC20** token on the zkEVM network. The method returns an object which has various other methods.

```js
const erc20token = hermezClient.erc20(<token address> , <isRoot>);
```

Passing second argument for `isRoot` is optional.

### For Child Token

Token on the zkEVM network can be initiated by using this syntax:

```js
const childERC20Token = hermezClient.erc20(<child token address>);
```

### For Root Token

Token on ethereum can be initiated by providing the **second parameter value as `true`**.

```js
const rootERC20Token = hermezClient.erc20(<root token address>, true);
```

## Check Balance

You can use the `getBalance` method to get the balance of a user account. It is available for both child and root token.

```js
// get balance of user
const balance = await erc20Token.getBalance(<user Address>);
```

## Approve Methods

### approve

> `approve` method can be used to approve required amount on the root token. It is required in order to deposit amount on the zkEVM network.

Reviewer: this is for both root and child token

> `approve` method can be used to approve required amount on the root token. It is required in order to deposit amount on the zkEVM network.

Author: ```suggestion
The `approve` method can approve the required amount on the root and child token. It is needed in order to deposit the amount on the zkEVM network.
```

```js
const erc20Token = hermezClient.erc20(<root token address>, true); // root token

// approve 1000 amount
const result = await erc20Token.approve(1000);

const txHash = await result.getTransactionHash();
const receipt = await result.getReceipt();
```

#### spenderAddress

The address on which approval is given is called the `spenderAddress`. It is a third-party user or a smart contract which can transfer your token on your behalf.

By default `spenderAddress` value is an ERC20 predicate address. You can specify `spenderAddress` value manually.

```js
// approve 1000 amount
const result = await erc20Token.approve(1000, {
    spenderAddress: <spender address value>
});
```

### approveMax

> `approveMax` method can be used to approve maximum amount on the root token.

Reviewer: this is also for both tokens, root and child

Reviewer: this is same everywhere in approval, allowance, isApprovalNeeded

> `approveMax` method can be used to approve maximum amount on the root token.

Author: ```suggestion
The `approveMax` method can approve the maximum amount on the root and child tokens.
```

```js
const erc20Token = hermezClient.erc20(<root token address>, true); // root token

const result = await erc20Token.approveMax();

const txHash = await result.getTransactionHash();
const receipt = await result.getReceipt();
```

#### spenderAddress

You can specify `spenderAddress` value manually.

```js
// approve 100 amount
const result = await erc20Token.approveMax({
    spenderAddress: <spender address value>
});
```

### isApprovalNeeded

> `isApprovalNeeded` checks if an approval is needed for the root token.

Author: ```suggestion
`isApprovalNeeded` checks if approval is needed for the root or child token.
```
